backend/api/main.py

```
@@ -268,6 +276,46 @@ async def list_sessions():
+        raise HTTPException(status_code=400, detail=e.dict())
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=f"결제 준비 중 오류가 발생했습니다: {str(e)}")
+
+@app.post("/payment/approve", response_model=PaymentApproveResponse)
+async def payment_approve(approve_request: PaymentApproveRequest):
+    """카카오페이 결제 승인 API"""
+    try:
+        response = await kakao_pay_service.approve_payment(approve_request)
+        return response
+    except PaymentErrorResponse as e:
+        raise HTTPException(status_code=400, detail=e.dict())
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)